chronosphere/intschema/generateintschema/main.go

```diff
@@ -397,8 +422,21 @@ func (f *fileData) loadType(
 		if typeName, ok := sharedElemTypeNames[t]; ok {
 			return typeInfo{name: typeName}
 		}
+		if recursiveFields[tfName] {
+			if tfName == inRecursiveField {
+				// End recursive type.
+				return typeInfo{name: parentTypeName}
+			} else if inRecursiveField != "" {
+				// TODO: while building inRecursiveField, we encountered a
+				// new recursive field (t). This is an artificial limitation
+				// of this code which can be removed.
+				panic("recursive types cannot contain other recursive types")
+			}
+			// Begin recursive type.
+			inRecursiveField = tfName
```

Comment:
instead of `recursiveFields` being a `map[string]bool` could it be `map[string]*schema.Schema` where the schema must match before we enter? We can still use the key existence to determine when to end (since that schema won't match)